lib/broccoli/ember-app.js

```diff
@@ -530,15 +531,63 @@ class EmberApp {
       });
     }
 
-    this.project.addons = this.project.addons.filter(function(addon) {
+    this.project.addons = this.project.addons.filter(addon => {
       if (this.shouldIncludeAddon(addon)) {
         if (addon.included) {
           addon.included(this);
         }
 
         return addon;
       }
-    }, this);
+    });
+  }
+
+  /**
+    Calls the importTransforms hook on addons.
+
+    @private
+    @method _importAddonTransforms
+  */
+  _importAddonTransforms() {
+
+    this.project.addons.forEach(addon => {
+      if (this.shouldIncludeAddon(addon)) {
+        if (addon.importTransforms) {
+          let transforms = addon.importTransforms();
+
+          if (!transforms) {
+            throw new Error(`Addon "${addon.name}" did not return a transform map from importTransforms function`);
+          }
+
+          Object.keys(transforms).forEach(transformName => {
+            const transformIndex = this._customTransformOrder.indexOf(transformName);
+            if (transformIndex > -1) {
+              // there is already a transform with a same name, therefore we warn the user
+              this.project.ui.writeWarnLine(`Addon "${addon.name}" is defining a transform name: ${transformName} that is already being defined. Using transform from addon: "${addon.name}".`);
+              this._customTransformOrder[transformIndex] = transformName;
+            } else {
+              // store the transform order to reterieve later
+              this._customTransformOrder.push(transformName);
+            }
+
+            let transformConfig = {
+              files: [],
+              options: {},
+            };
+
+            // store the transform info
+            if (typeof transforms[transformName] === 'object') {
+              transformConfig['callback'] = transforms[transformName].transform;
+              transformConfig['processOptions'] = transforms[transformName].processOptions;
+            } else {
```

Comment:
Should we have this be an `} else if (typeof transforms[transformName] === 'function') {}` then add a "real" `else` clause that throws?